platform/frontend_connectors/schema_transformer.go

```diff
@@ -1185,3 +1186,26 @@ func (s *SchemaCheckPass) applyMatchOperator(indexSchema schema.Schema, query *m
 	return query, nil
 
 }
+
+func (s *SchemaCheckPass) applyClusterFunction(currentSchema schema.Schema, query *model.Query) (*model.Query, error) {
+	if s.cfg.ClusterName == "" {
+		return query, nil
+	}
+	visitor := model.NewBaseVisitor()
+	table, ok := s.tableDiscovery.TableDefinitions().Load(query.TableName)
+	if !ok {
+		return nil, fmt.Errorf("table %s not found", query.TableName)
+	}
+	if !table.ExistsOnAllNodes {
+		return query, nil
+	}
+	visitor.OverrideVisitTableRef = func(b *model.BaseExprVisitor, e model.TableRef) interface{} {
+		return model.NewFunction("cluster", model.NewLiteral(s.cfg.ClusterName), e)
+	}
+	logger.Info().Msgf("applyClusterFunction: %s", s.cfg.ClusterName)
```

Comment:
```suggestion
	logger.Debug().Msgf("applyClusterFunction: %s", s.cfg.ClusterName)
```
In some setups this could get quite noisy I think